### PR TITLE
Update to change how citations are handled and add scrolling to tag lists

### LINKED ISSUE #46: please add consistent formatting to the citation field for all entries
For example : Title, Authors, Venue, Year
Long-Term Recurrent Convolutional Networks for Visual Recognition and Description, Jeffrey Donahue, Lisa Anne Hendricks, Sergio Guadarrama, Marcus Rohrbach, Subhashini Venugopalan, Kate Saenko, Trevor Darrell; The IEEE Conference on Computer Vision and Pattern Recognition (CVPR), 2015

### PR DESCRIPTION
**Pull request by ddehueck**
Resolves #46 

**Note**: @haroldfry  will need to update the current citation column in the DB to a new name ("old_citation"). Before updating database to the model defined in this pull request. **Otherwise, we will lose all current citation data.**

**Renaming `citation` field to `old_citation` - _Important_!**

1.) Run a migration generation with changing anything e.g. `alembic revision --autogenerate -m “renaming_citation_field”` - should have empty functions

2.) Open generated file in alembic/versions

3.) Add the following line in the upgrade function: `op.alter_column('datasets', 'citation', new_column_name='old_citation')`

The upgrade function should now look like:
```
def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.alter_column('datasets', 'citation', new_column_name='old_citation')
    pass
    # ### end Alembic commands #
```

5.) Run the migration: `alembic upgrade head` - check the database to see if `citation` is now `old_citation`

5.) Finally, rename the citation variable in datasets/models to old_citation 

6.) Then upgrade to datasets/model file that has the new citation fields - defined in this pull request

### CODE REVIEW

**Comment by ddehueck** [comment]
@Eric4Jiang Could you check the changes to the elasticsearch configuration docs for correctness.

File: backend/app/datasets/routes.py [hunk 1]
```diff
@@ -111,10 +111,6 @@ def post(self):
         # Holds a list of tag ids to be associated with the dataset.
         tags = req_body.pop('tags', [])
 
-        # Ensure that new datasets have citations
-        if req_body.get('citation') is None:
-            return {'errors': 'Citation missing'}
-
```
Comment (Eric4Jiang) [inline backend/app/datasets/routes.py]: Why was this removed? Is it fixed to be handled in the frontend?

File: backend/app/datasets/models.py [hunk 2]
```diff
@@ -34,6 +33,15 @@ class Dataset(SearchableMixin, db.Model):
...
 
+    # Citation Fields
+    cite_title = db.Column(db.String(1000), nullable=True)
+    cite_authors = db.Column(db.String(1000), nullable=True)
+    cite_venue = db.Column(db.String(1000), nullable=True)
+    cite_year = db.Column(db.Integer, nullable=True)
+
+    # Old Citation Field
+    old_citation = db.Column(db.String(1000), nullable=True)
+
```
Comment (Eric4Jiang) [inline backend/app/datasets/models.py]: Should we allow datasets to be uploaded with all null values for citations? I thought we wanted every dataset to require it.

Comment (ddehueck) [reply backend/app/datasets/models.py]: I think we ran into some error when we uploaded some manually and had to change it to nullable. The requirement it enforced on the frontend though.

File: backend/app/datasets/routes.py (hunk 1 above)
Comment (ddehueck) [reply backend/app/datasets/routes.py]: It should be handled in the frontend.

File: backend/app/datasets/models.py (hunk 2 above)
Comment (Eric4Jiang) [reply backend/app/datasets/models.py]: The error came from trying to migrate over old datasets that did not have any citation, and therefore gave an error because nullable was set to false. Ideally I think we would like to keep billable = false, we just need another way to handle the old datasets without citation

**Comment by ddehueck** [comment]
@Eric4Jiang Any chance you were able to get this working?

**Comment by Eric4Jiang** [comment]
I'm checking it now. Sorry for the delay

**Comment by ddehueck** [comment]
@haroldfry Before pushing this new code to production the database should probably be backed up or else we may lose all current citation data if this update goes wrong.

**Comment by Eric4Jiang** [comment]
Just to be more elaborate on your point 3. Once you clone the pull request and autogenerate the migration, the upgrade function will look like:

```def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.add_column('datasets', sa.Column('cite_authors', sa.String(length=1000), nullable=True))
    op.add_column('datasets', sa.Column('cite_title', sa.String(length=1000), nullable=True))
    op.add_column('datasets', sa.Column('cite_venue', sa.String(length=1000), nullable=True))
    op.add_column('datasets', sa.Column('cite_year', sa.Integer(), nullable=True))
    op.add_column('datasets', sa.Column('old_citation', sa.String(length=1000), nullable=True))
    op.drop_column('datasets', 'citation')
    # ### end Alembic commands ###
```

It's important to REPLACE 
```    
    op.add_column('datasets', sa.Column('old_citation', sa.String(length=1000), nullable=True))
    op.drop_column('datasets', 'citation')
```
with 
```op.alter_column('datasets', 'citation', new_column_name='old_citation')```
so that the final upgrade functions looks like 
```def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.add_column('datasets', sa.Column('cite_authors', sa.String(length=1000), nullable=True))
    op.add_column('datasets', sa.Column('cite_title', sa.String(length=1000), nullable=True))
    op.add_column('datasets', sa.Column('cite_venue', sa.String(length=1000), nullable=True))
    op.add_column('datasets', sa.Column('cite_year', sa.Integer(), nullable=True))
    op.alter_column('datasets', 'citation', new_column_name='old_citation')
    # ### end Alembic commands ###
```

Otherwise, the 'citation' column will be dropped.

**Comment by ddehueck** [comment]
@Eric4Jiang Yes that works. I did it in two steps: 1. Alter citation column name to old_citation 2. Add new columns for new citation format.
